spec/controllers/samples_controller_spec.rb

```diff
@@ -0,0 +1,98 @@
+require 'rails_helper'
+
+RSpec.describe SamplesController, type: :controller do
+  create_users
+
+  pipeline_stages = [{
```

Comment:
you could also call this one pipeline_run_stages_data to clarify that they're not pipeline stage objects